Creates *advanced* SMES coils

juxjux9930 opened the pull request:
🆑 
rscadd: Creates the "advanced" SMES coil for admin and away site use.
tweak: Skrell and Ascent SMES units now use advanced SMES coils.
/🆑

Power management is a pain, and there are also few tangible rewards from some away sites. Two birds with one stone here - the new coils will make mapping high-power sites require less SMES dumping, and could be mapped into away sites as a sort-of "reward" exploration could bring back for engineering. The new coil has double the I/O of transmission coils and double the storage of capacity coils. It's fairly meaty.

I've also added a SMES preset that uses the new coils, for your mapping pleasures.

juxtaposed9930 commented:
Didn't have an empty line at the end of one of smes_presets, fixed it.

Shadowtail117 commented:
Errors:
```
## ERROR: Couldn't find atom path specified in map: /obj/machinery/power/smes/buildable/power_shuttle/ascent{
## ERROR: Couldn't find atom path specified in map: /obj/machinery/power/smes/buildable/power_shuttle/ascent{
## ERROR: Couldn't find atom path specified in map: /obj/machinery/power/smes/buildable/power_shuttle/ascent{
## ERROR: Couldn't find atom path specified in map: /obj/machinery/power/smes/buildable/power_shuttle/ascent{
## ERROR: Couldn't find atom path specified in map: /obj/machinery/power/smes/buildable/power_shuttle/ascent{
```
You changed the path of the ascent SMES so you need to reflect that in the map file as well.

juxtaposed9930 commented:
Changed it in the map, don't know where that'd coming from, which is why I'm looking.

juxtaposed9930 commented:
Ah, the map didn't save.

Spookerton commented on maps/away/skrellscoutship/skrellscoutship.dm:
```diff
@@ -191,8 +191,7 @@
 	uncreated_component_parts = list(
-		/obj/item/weapon/stock_parts/smes_coil/super_io = 2,
-		/obj/item/weapon/stock_parts/smes_coil/super_capacity = 2)
+		/obj/item/weapon/stock_parts/smes_coil/advanced = 2,)
```
```suggestion
		/obj/item/weapon/stock_parts/smes_coil/advanced = 2
	)
```

Spookerton commented on code/modules/power/smes_presets.dm:
```diff
@@ -22,3 +22,13 @@
+
+/obj/machinery/power/smes/buildable/preset/admin
+	uncreated_component_parts = list(
+		/obj/item/weapon/stock_parts/smes_coil/advanced = 4)
```
```suggestion
		/obj/item/weapon/stock_parts/smes_coil/advanced = 4
	)
```

Spookerton commented on code/modules/ascent/ascent_machines.dm:
```diff
@@ -271,7 +278,7 @@ MANTIDIFY(/obj/machinery/door/airlock/external/bolted, "mantid airlock", "door")
 	color = COLOR_VIOLET
 	construct_state = null
-	
+
```
```suggestion
```

Spookerton commented on code/modules/ascent/ascent_machines.dm:
```diff
@@ -237,12 +237,19 @@ MANTIDIFY(/obj/machinery/door/airlock/external/bolted, "mantid airlock", "door")
 	overlay_icon = 'icons/obj/machines/power/mantid_smes.dmi'
-	construct_state = /decl/machine_construction/default/no_deconstruct
+	uncreated_component_parts = list(
+		/obj/item/weapon/stock_parts/smes_coil/advanced = 2,
+		/obj/item/weapon/stock_parts/smes_coil/super_capacity = 2)
```
```suggestion
		/obj/item/weapon/stock_parts/smes_coil/advanced = 2
	)
```

Spookerton reviewed (changes requested):
in the mapped instances in the diff,
```
/obj/machinery/power/smes/buildable/preset/ascent{
	charge = 5e+006
	},
```
to just
```
/obj/machinery/power/smes/buildable/preset/ascent,
```

\-- No point leaving in the mapped vars, since they're `_fully_charged = TRUE`

Spookerton commented on code/modules/ascent/ascent_machines.dm:
```diff
@@ -237,12 +237,19 @@ MANTIDIFY(/obj/machinery/door/airlock/external/bolted, "mantid airlock", "door")
-	construct_state = /decl/machine_construction/default/no_deconstruct
+	uncreated_component_parts = list(
+		/obj/item/weapon/stock_parts/smes_coil/advanced = 2
+			)
```
```suggestion
	)
```

Spookerton commented on code/modules/power/smes_presets.dm:
```diff
@@ -22,3 +22,14 @@
+/obj/machinery/power/smes/buildable/preset/admin
+	uncreated_component_parts = list(
+		/obj/item/weapon/stock_parts/smes_coil/advanced = 4
+			)
```
```suggestion
	)
```

Spookerton replied:
eeee

Spookerton commented on maps/away/skrellscoutship/skrellscoutship.dm:
```diff
@@ -191,8 +191,8 @@
-		/obj/item/weapon/stock_parts/smes_coil/super_io = 2,
-		/obj/item/weapon/stock_parts/smes_coil/super_capacity = 2)
+		/obj/item/weapon/stock_parts/smes_coil/advanced = 2
+			)
```
```suggestion
	)
```

Spookerton replied:
a liieeeee

Spookerton reviewed (changes requested):
screaming

juxtaposed9930 commented:
Me squash

juxjux9930 committed:
power to the smes


Update code/modules/ascent/ascent_machines.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/power/smes_presets.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update maps/away/skrellscoutship/skrellscoutship.dm

Co-authored-by: Spookerton <spkrtn@pm.me>

juxtaposed9930 commented:
fixeded it Spook

Spookerton reviewed (approved):
Deliciously correct list form.
